stable/database/templates/statefulset.yaml

```diff
@@ -686,6 +698,10 @@ spec:
         matchLabels:
           database: {{ .Values.database.name }}
 {{- end }}
+      {{- if .Values.database.persistence.dataSourceRef }}
```

Comment:
I think we should conditionalize on the name being non-empty and have an entire `dataSourceRef` section in our `values.yaml` with the following default values:

```yaml
      dataSourceRef:
        name: ""
        kind: VolumeSnapshot
        apiGroup: snapshot.storage.k8s.io
```

This way someone only has to specify `database.persistence.dataSourceRef.name` to some value to refer to VolumeSnapshot in the same namespace.

Reply:
I like this idea. Will do.